controllers/search.js

```diff
@@ -12,22 +11,33 @@ module.exports = {
   query: function query(req, res) {
     var model = new Model(req.body);
 
-    return model.read().then(function resolved(result) {
-
-      var records = result.records;
-      model.set('records', records);
-      model.set('query', req.body);
-
-      req.session.model = model;
-
-      if (records.length === 1) {
-        res.redirect('/details');
-      } else {
-        res.redirect('/results');
-      }
-
-    }).catch(function rejected(err) {
-      throw new Error(err);
-    });
+    return model
+      .read()
+      .then(function resolved(result) {
+        var records = result.records;
+        model.set('records', records);
+        model.set('query', req.body);
+
+        req.session.model = model;
+
+        if (records.length === 1) {
+          res.redirect('/details');
+        } else {
+          res.redirect('/results');
+        }
+      }, function rejected(err) {
+        switch (err.name) {
+          case 'NotFoundError':
+            model.set('records', null);
+            model.set('query', req.body);
+            req.session.model = model;
+
+            res.redirect('/results');
+            break;
+          default:
+            throw (err instanceof(Error)) ? err
+                                          : new Error(err);
+        }
```

Comment:
I don't think a `switch is really necessary. I understand that there may well be more error types to handle later, but we can handle that when it happens. Also, it's just a my personal preference, but I think this reads better:

```js
if (err.name === 'NotFoundError') {
  model.set('records', null);
  model.set('query', req.body);
  req.session.model = model;

  return res.redirect('/results');
}

throw (err instanceof(Error)) ? err : new Error(err);
```